Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -442,12 +442,17 @@ - (void)setValuesForKeysWithDictionary:(NSDictionary*)keyedValues {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSDictionary*)dictionaryWithValuesForKeys:(NSArray*)keys {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSDictionary* results = [NSMutableDictionary dictionaryWithCapacity:keys.count];
+
+    for (NSString* key in keys) {
+        id retValue = [self valueForKey:key];
+        [results setValue:(retValue ?: [NSNull null]) forKey:key];
+    }
+
+    return [[results copy] autorelease];
```

Comment:
this seems like it may be better served by a vector or something local so that you can just make the final dictionary without doing potentially expensive grows and copies on the mutable guy.